In [ ]:
#Q1 import the libraries
import pandas as pd
import scipy 
import numpy as np
from scipy import stats
import seaborn as sns

In [ ]:
from google.colab import files
files.upload()

Saving Cutlets.csv to Cutlets.csv


{'Cutlets.csv': b'Unit A,Unit B\r\n6.809,6.7703\r\n6.4376,7.5093\r\n6.9157,6.73\r\n7.3012,6.7878\r\n7.4488,7.1522\r\n7.3871,6.811\r\n6.8755,7.2212\r\n7.0621,6.6606\r\n6.684,7.2402\r\n6.8236,7.0503\r\n7.393,6.881\r\n7.5169,7.4059\r\n6.9246,6.7652\r\n6.9256,6.038\r\n6.5797,7.1581\r\n6.8394,7.024\r\n6.597,6.6672\r\n7.2705,7.4314\r\n7.2828,7.307\r\n7.3495,6.7478\r\n6.9438,6.8889\r\n7.156,7.422\r\n6.5341,6.5217\r\n7.2854,7.1688\r\n6.9952,6.7594\r\n6.8568,6.9399\r\n7.2163,7.0133\r\n6.6801,6.9182\r\n6.9431,6.3346\r\n7.0852,7.5459\r\n6.7794,7.0992\r\n7.2783,7.118\r\n7.1561,6.6965\r\n7.3943,6.578\r\n6.9405,7.3875\r\n'}

In [ ]:
cut=pd.read_csv("Cutlets.csv")
cut.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [ ]:
cut.mean()

Unit A    7.019091
Unit B    6.964297
dtype: float64

In [ ]:
cutnew = cut.sample(5)
cutnew

Unit A  Unit B
3   7.3012  6.7878
26  7.2163  7.0133
17  7.2705  7.4314
16  6.5970  6.6672
34  6.9405  7.3875

In [ ]:
cutnew.mean()

Unit A    7.06510
Unit B    7.05744
dtype: float64

In [ ]:
cutnew.std()

Unit A    0.298093
Unit B    0.344865
dtype: float64

In [ ]:
n1=5
n2=5
x1bar=7.04314
x2bar=6.93726
sd1=0.294960
sd2=0.276628
alpha = 0.05

In [ ]:
SE = np.sqrt(sd1**2/n1+sd2**2/n2)
t = ((x1bar-x2bar)/SE)
print("calculate value of t test is: ",t)

calculate value of t test is:  0.5854739597300976


In [ ]:
df=n1+n2-2
cv=stats.t.ppf(alpha,df)
print("critical value is: ",cv)

critical value is:  -1.8595480375228428


In [ ]:
pval=stats.t.cdf(t,df)
print("p value of right-sided test is: ",pval)

p value of right-sided test is:  0.7128156108298787


We can clearly see that there is a significant difference between p value = 0.7128156108298787 and t tests value = 0.5854739597300976

In [ ]:
#Q2 LABTAB
from google.colab import files
files.upload()

Saving LabTAT.csv to LabTAT.csv


{'LabTAT.csv': b'Laboratory 1,Laboratory 2,Laboratory 3,Laboratory 4\r\n185.35,165.53,176.7,166.13\r\n170.49,185.91,198.45,160.79\r\n192.77,194.92,201.23,185.18\r\n177.33,183,199.61,176.42\r\n193.41,169.57,204.63,152.6\r\n179.45,197,181.51,161.12\r\n191.37,166.36,214.21,154.02\r\n166.81,169.6,183.43,163.25\r\n158.81,175.36,191.6,152.79\r\n165.88,198.68,208.43,161.98\r\n174.75,189.12,231,171.22\r\n193.37,140.55,198.37,183.67\r\n184.75,160.44,226.62,142.95\r\n178.54,167.03,214.44,152.37\r\n180.19,182.67,159.69,163.81\r\n172.17,155.73,214.14,156.06\r\n172,183.07,212.29,176.44\r\n184.92,177.7,209.25,173.68\r\n187.38,191.62,198.54,161.49\r\n194.17,186.85,229.02,175.05\r\n192.03,161.09,214.71,150.22\r\n193.77,173.01,172.27,148.17\r\n178.24,199.43,205.31,166.21\r\n171.41,185.67,180.35,145.4\r\n178.3,182.69,214.58,166.42\r\n176.13,180.45,231.76,132.21\r\n180.67,182.37,215.78,143.26\r\n177.38,189.99,201.78,154.06\r\n190.36,167.55,191.64,179.08\r\n199.99,190.09,217.56,144.82\r\n181.36,168.01,193

In [ ]:
lab=pd.read_csv("LabTAT.csv")
lab.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [ ]:
alpha = 0.05  
# Perfrom One Way ANOVA
outcome = stats.f_oneway(lab['Laboratory 1'],lab['Laboratory 2'],lab['Laboratory 3'],lab['Laboratory 4'])
# gettin p-value of test
p_value= outcome[1]

In [ ]:
p_value

2.1156708949992414e-57

In [ ]:
#Null hypothesis mu_1 = mu_2 = mu_3 = mu_4
if p_value <= alpha:
    print ('Null hypothesis is unlikely to except')
else:
   print('Null hypothesis cannot be rejected')

Null hypothesis is unlikely to except


In [ ]:
#Q3
from google.colab import files
files.upload()

Saving BuyerRatio.csv to BuyerRatio.csv


{'BuyerRatio.csv': b'Observed Values,East,West,North,South\r\nMales,50,142,131,70\r\nFemales,435,1523,1356,750\r\n'}

In [ ]:
df = pd.read_csv("BuyerRatio.csv")
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
arr = np.array([[50,142,131,70],[435,1523,1356,750]])
arr

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [ ]:
from scipy.stats import chi2_contingency
chi2_contingency(arr)

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [ ]:
# significance level
alpha = 0.05  
# Perfrom One Way ANOVA
outcome = stats.f_oneway(df['East'],df['West'],df['North'],df['South'])
# gettin p-value of test
p_value= outcome[1]

In [ ]:
#Null hypothesis mu_1 = mu_2 = mu_3
if p_value < alpha:
    # we reject null hypothesis
    print ('ALL proportion are Equal i.e. We reject Null hypothesis')
else:
    # we reject alternative hypothesis
    print('Not all proportion are Equal - We accept the Null hypothesis')

Not all proportion are Equal - We accept the Null hypothesis


In [ ]:
#Q4
from google.colab import files
files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm.csv


{'Costomer+OrderForm.csv': b'Phillippines,Indonesia,Malta,India\r\nError Free,Error Free,Defective,Error Free\r\nError Free,Error Free,Error Free,Defective\r\nError Free,Defective,Defective,Error Free\r\nError Free,Error Free,Error Free,Error Free\r\nError Free,Error Free,Defective,Error Free\r\nError Free,Error Free,Error Free,Error Free\r\nError Free,Defective,Error Free,Error Free\r\nError Free,Error Free,Error Free,Error Free\r\nError Free,Error Free,Error Free,Error Free\r\nError Free,Error Free,Error Free,Error Free\r\nDefective,Error Free,Error Free,Error Free\r\nError Free,Error Free,Error Free,Error Free\r\nError Free,Error Free,Error Free,Defective\r\nError Free,Error Free,Error Free,Error Free\r\nError Free,Error Free,Error Free,Error Free\r\nError Free,Error Free,Defective,Error Free\r\nError Free,Error Free,Error Free,Error Free\r\nError Free,Error Free,Error Free,Error Free\r\nError Free,Error Free,Error Free,Error Free\r\nDefective,Error Free,Error Free,Defective\r\nErro

In [ ]:
data =pd.read_csv("Costomer+OrderForm.csv")
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [ ]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [ ]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [ ]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [ ]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [ ]:
arr = np.array([[271,267,269,280],[29,33,31,20]])
arr

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [ ]:
from scipy.stats import chi2_contingency
chi2_contingency(arr)

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

In [ ]:
#One Way ANOVA Example


# creating data frame 
d = {'Phillippines': pd.Series([271,29]),
     'Indonesia': pd.Series([267,33]),
     'Malta': pd.Series([269,31]),
     'India': pd.Series([280,20])}

df = pd.DataFrame(d)

# significance level
alpha = 0.05  
# Perfrom One Way ANOVA
outcome = stats.f_oneway(df['Phillippines'],df['Indonesia'],df['Malta'],df['India'])
# gettin p-value of test
p_value= outcome[1]




In [ ]:
#Null hypothesis mu_1 = mu_2 = mu_3
if p_value <= alpha:
    # we reject null hypothesis
    print ('Null hypothesis is unlikely to except')
else:
    # we reject alternative hypothesis
    print('Null hypothesis cannot be rejected')

Null hypothesis cannot be rejected
